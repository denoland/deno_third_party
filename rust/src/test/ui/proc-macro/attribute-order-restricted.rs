// aux-build:attr_proc_macro.rs

extern crate attr_proc_macro;
use attr_proc_macro::*;

#[attr_proc_macro] // OK
#[derive(Clone)]
struct Before;

#[derive(Clone)]
#[attr_proc_macro] //~ ERROR macro attributes must be placed before `#[derive]`
struct After;

fn main() {}
