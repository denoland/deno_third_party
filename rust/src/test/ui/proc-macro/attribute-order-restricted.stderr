error: macro attributes must be placed before `#[derive]`
  --> $DIR/attribute-order-restricted.rs:11:1
   |
LL | #[attr_proc_macro] //~ ERROR macro attributes must be placed before `#[derive]`
   | ^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

